api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java

```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
         registerLocally(nationalBiometricSubject, fingerScan);
     }
 
+    @Override
+    public void synchronizeFingerprints(M2SysCaptureResponse fingerScan, FingerScanStatus fingerScanStatus) {
+        if (fingerScanStatus.isRegisteredNationally()) {
+            if (patientHelper.findByNationalFpId(fingerScanStatus.getNationalBiometricSubject().getSubjectId()) == null) {
+                attachNationalIdToThePatient(fingerScanStatus.getLocalBiometricSubject().getSubjectId());
+            }
+        } else {
+            registerNationally(fingerScanStatus.getLocalBiometricSubject(), fingerScan);
+        }
+    }
+
+    private void attachNationalIdToThePatient(String localId) {
```

Comment:
I think that you should pass Patient and nationalId here or name this method more appropriately. It is a bit confusing and difficult to understand.

Reply:
Yes, I don't get why are we saving the localID as the national ID